truss/templates/base.Dockerfile.jinja

```diff
@@ -5,8 +5,8 @@ ENV PYTHON_EXECUTABLE {{ config.base_image.python_executable_path or 'python3' }
 
 {% block fail_fast %}
 RUN grep -w 'ID=debian\|ID_LIKE=debian' /etc/os-release || { echo "ERROR: Supplied base image is not a debian image"; exit 1; }
-RUN $PYTHON_EXECUTABLE -c "import sys; sys.exit(0) if sys.version_info.major == 3 and sys.version_info.minor >=8 and sys.version_info.minor <=11 else sys.exit(1)" \
-    || { echo "ERROR: Supplied base image does not have 3.8 <= python <= 3.11"; exit 1; }
+RUN $PYTHON_EXECUTABLE -c "import sys; sys.exit(0) if sys.version_info.major == 3 and sys.version_info.minor >=8 and sys.version_info.minor <=12 else sys.exit(1)" \
+    || { echo "ERROR: Supplied base image does not have 3.8 <= python <= 3.12"; exit 1; }
```

Comment:
Do we officially not support 3.12? What would be needed? Are there anywhere else in the code base constants or references to supported versions that need to be updated? Could we ideally move this value to one single source of truth place and then provide it as a variable to filling this jinja template?

Reply:
@squidarth can provide more context here.

Reply:
@marius-baseten we don't have 3.12 in any of the Baseten images (see https://github.com/basetenlabs/truss/blob/a121e69bdb4a5519a19a78aa2bc16c6672c82ec1/truss/truss_config.py#L521), the only way to use python 3.12 is via custom base image. This line enables 3.12 in custom base images.

it's a little bit of a subtle distinction, but this change has fairly limited blast radius. let's go with this for now, we should follow-up by enabling 3.12 more broadly & having that be available as a baseten image, and when we do that we can clean up some of this

Reply:
Ok the 12/11 question is fine. But still it would be better to manage these version from one constant rather than as various literals scattered throughout the repo.

Reply:
Ok taht makes sense, @tianshuc0731 , could you add 3.12 and 3.8 as the MAXIMUM and MINIMUM python versions in this contants file: https://github.com/basetenlabs/truss/blob/a121e69bdb4a5519a19a78aa2bc16c6672c82ec1/truss/constants.py#L8, and then pass them into the template?

I think that's the last thing here, happy to jump on a zoom if that would be helpful

Reply:
done